music21/romanText/tsvConverter.py

```diff
@@ -0,0 +1,670 @@
+            else:  # entry.measure = previousMeasure + 1
+                m = stream.Measure(number=entry.measure)
+                m.offset = entry.totbeat
+                p.insert(m)
+                if entry.timesig != currentTimeSig:
+                    newTS = meter.TimeSignature(entry.timesig)
+                    m.insert(entry.beat - 1, newTS)
+
+                    currentTimeSig = entry.timesig
+                    currentMeasureLength = newTS.barDuration.quarterLength
+
+                previousMeasure = entry.measure
+                currentOffset = entry.totbeat
+
+        s.append(p)
+
+        self.prepdStream = s
+
+        return s
+
+# ------------------------------------------------------------------------------
+
+class M21toTSV:
+    '''
+    Exports a music21 stream to tabular data and (optionally) writes the file.
+    '''
+
+    def __init__(self, m21Stream):
+        self.m21Stream = m21Stream
+        self.tsvData = self.m21ToTsv()
+
+    def m21ToTsv(self):
+        '''
+        Converts a list of music21 chords to a list of lists
+        which can then be written to a tsv file with toTsv(), or processed another way.
+        '''
+
+        tsvData = []
+
+        for thisEntry in self.m21Stream.recurse():
+            if 'RomanNumeral' in thisEntry.classes:
+
+                relativeroot = None
+                if thisEntry.secondaryRomanNumeral:
+                    relativeroot = thisEntry.secondaryRomanNumeral.figure
+
+                altChord = None
+                if thisEntry.secondaryRomanNumeral:
+                    if thisEntry.secondaryRomanNumeral.key == thisEntry.key:
+                        altChord = thisEntry.secondaryRomanNumeral.figure,
+
+                thisInfo = [thisEntry.figure,  # NB: slightly different from DCML: no key.
+                            altChord,
+                            thisEntry.measureNumber,
+                            thisEntry.beat,
+                            None,  # 'totbeat'
+                            thisEntry.getContextByClass('TimeSignature').ratioString,
+                            self.m21Stream.metadata.opusNumber,
+                            self.m21Stream.metadata.number,
+                            self.m21Stream.metadata.movementNumber,
+                            thisEntry.quarterLength,
+                            None,  # 'global_key',
+                            thisEntry.key,  # 'local_key',
+                            None,  # 'pedal',
+                            thisEntry.romanNumeralAlone,  # numeral
+                            None,  # 'form',
+                            thisEntry.figuresWritten,  # figbass
+                            None,  # 'changes',
+                            relativeroot,
+                            None,  # 'phraseend'
+                            ]
+                tsvData.append(thisInfo)
+
+        return tsvData
+
+    def write(self, outFilePath='./', outFileName='TSV_FILE.tsv',):
```

Comment:
file path and file name is generally sent as one parameter.  Let the user make the filepath for themselves.